# GRASP GENERATION PARAMS
object_key: mount2
stp_id: pose_0
num_grasps: 10
debug: 0

delta_x: 0.154
delta_y: -0.077
delta_z: -0.038

cb_platform_x: 0.527
cb_platform_y: -0.011
cb_platform_z: -0.144

delta_pregrasp: 0.2
delta_lift: 0.2
robot_base_angle_thresh: 0.785
table_clearance: 0.025
angle_from_table: 0.125
force_closure_thresh: 1

output_dir: /Users/jeff/Google Drive/fanuc_grasps/mount2/test

# Parameters of the metrics to output
deterministic_grasp_metrics:
  - force_closure
  - ferrari_canny_L1
robust_grasp_metrics:
  - force_closure
  - ferrari_canny_L1
uncertainty_level: low

# Core database parameters
dataset_counts: data/counts.txt
datasets:
  dexnet_physical_experiments:
    start_index: 0
    end_index: 9000

max_chunk_size: 1

database_dir: &id_database_dir data/db
database_cache_dir: *id_database_dir
database_name: physical_experiment_db.hdf5
results_database_name: results.hdf5

# PLOTTING
line_width: 6
alpha: 0.15
plot_dim: 0.6

# OLD PARAMS
num_trials: 0
sleep_time: 10

priors_dataset: keys_10000_train_and_val
priors_feature_names:
  - nearest_features_1000_train_and_val

use_stable_poses: 1
image_type: segmask

gripper: fanuc_lehf

# Registration configuration
registration:
  table_front_depth: 0.0
  table_rear_depth:  0.9
  table_surface_tol: 0.0025

  focal_length: 525
  index_im_dim: 256
  depth_im_crop_dim: 400
  depth_im_median_filter_dim: 9
  depth_im_erosion_filter_dim: 3
  
  cache_im_filename: data/cnn_grayscale_image.jpg
  calibration_dir: data/calibration

  num_nearest_neighbors: 1
  icp_sample_size: 100
  icp_relative_point_plane_cost: 100
  icp_regularization_lambda: 0.01
  feature_matcher_dist_thresh: 0.05
  feature_matcher_norm_thresh: 0.75
  num_registration_iters: 5

  font_size: 15

# Caffe configuration
caffe:
  caffe_tools: '/home/jmahler/Libraries/caffe/build/tools/'
  config_dir: '/home/jmahler/mel/GPIS_data/caffe_data/'
  caffe_model: 'caffenet.caffemodel'
  caffe_finetuning_model: 'alexnet.caffemodel'
  deploy_file: 'deploy_conv5.prototxt'
  mean_file: 'mean.npy'

  portion_training: 1.00
  deploy_mode: 'gpu'
  batch_size: 10
  pooling_method: 'mean'

  rendered_image_ext: '.jpg'
  rendered_image_dir: '/home/jmahler/mel/GPIS_data/rendered_sets/all'

  images_per_object: 50
  raw_scale: 255
  image_dim_x: 256
  image_dim_y: 256

# Window weights #####################################################
inf:   &id_inf       100000000000000000000.0
zero:  &id_zero      0.0
weight_proj_win:     0.0
weight_grad_x:       gaussian_500.0_1.75
weight_grad_y:       gaussian_500.0_1.75
weight_curvature:    0.0
weight_grasp_center: 0.0
weight_grasp_axis:   0.0
weight_grasp_angle:  0.0
weight_gravity:      200.0

# Kernel params
kernel_sigma: 1.0
kernel_l: 1.0
kernel_tolerance: 0.001

prior_neighbor_weight: 0.001
prior_neighbor_distance: 20
prior_num_neighbors: 0
prior_kernel_tolerance: 0.001

lb_alpha: 0.50

# Params to run over
param_values:
  moment:
      - 50
      - 150
  num_nearest:
      - 10
  nn_weight:
      - 0.001

# GCE parameters #####################################################
client_secrets: &id_client_secrets client_secrets.json
oauth_storage:  &id_oauth_storage  oauth2.dat
disk_type:      &id_disk_type      PERSISTENT
num_processes: 8
project:        nth-clone-620
bucket:          dex-net-results-2 # TODO
compute:
  project:        &id_project        nth-clone-620
  bucket:         &id_bucket         dex-net-results-2 # TODO
  api_version:    v1
  machine_type:   n1-standard-1
  disk_size_gb: 2048
  zones:
    - asia-east1-a
    - us-central1-a
    - europe-west1-b
  instance_quota: 200
  startup_script: scripts/cm_startup.sh # TODO
  image_project:  *id_project
  image:          dex-net-image
  data_disks:
    - dex-net-data-2-big-asia
    - dex-net-data-2-big
    - dex-net-data-2-big-europe
  service_email:  default
  network:        default
  run_script:     src/grasp_selection/correlated_bandits_priors.py
  results_script: src/grasp_selection/compile_bandit_prior_results.py
  update_script: src/grasp_selection/compile_bandit_prior_results.py
  scopes:
    - https://www.googleapis.com/auth/compute
    - https://www.googleapis.com/auth/devstorage.full_control
    - https://www.googleapis.com/auth/devstorage.read_only
    - https://www.googleapis.com/auth/devstorage.read_write

# Notification parameters ############################################
gmail_user:     &id_gmail_user      RavenDebridement
gmail_password: &id_gmail_password  surgicallab
notify_email:   &id_notify_email    jmahl42@gmail.com